File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretCoordinateToPayload.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.config.persistence.split_secrets;

import com.fasterxml.jackson.databind.JsonNode;

/**
 * POJO to hold secret coordinate and the secret.
 *
 * @param secretCoordinate secret coordinate
 * @param payload the secret
 * @param secretCoordinateForDB json coordinate
 */
// todo (cgardens) - i don't understand these params.
public record SecretCoordinateToPayload(SecretCoordinate secretCoordinate,
                                        String payload,
                                        JsonNode secretCoordinateForDB) {

}

File: airbyte-config/config-secrets/src/main/kotlin/secrets/SecretCoordinateToPayload.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.config.secrets

import com.fasterxml.jackson.databind.JsonNode

/**
 * POJO to hold secret coordinate and the secret.
 *
 * @param secretCoordinate secret coordinate
 * @param payload the secret
 * @param secretCoordinateForDB json coordinate
 */
data class SecretCoordinateToPayload(
  val secretCoordinate: SecretCoordinate,
  val payload: String,
  val secretCoordinateForDB: JsonNode,
)
